adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,322 @@
+
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return resetDigitalRequest{}, err
+	}
+	if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+		return resetDigitalRequest{}, err
+	}
+	reqData.Imps[0].ZoneID.PlacementID = resetDigitalExt.PlacementID
+
+	return reqData, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response resetDigitalBidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+
+	var errs []error
+	requestImps := make(map[string]openrtb2.Imp)
+	for _, imp := range request.Imp {
+		requestImps[imp.ID] = imp
+	}
+
+	for i := range response.Bids {
+		resetDigitalBid := &response.Bids[i]
+
+		bid, err := getBidFromResponse(resetDigitalBid)
+		// handle the error
+		if bid == nil {
+			// it would be better to return an error here
+			errs = append(errs, err)
+			continue
+		}
+
+		bidType, err := GetMediaTypeForImp(requestImps, bid.ImpID)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		b := &adapters.TypedBid{
+			Bid:     bid,
+			BidType: bidType,
+			Seat:    openrtb_ext.BidderName(resetDigitalBid.Seat),
+		}
+		bidResponse.Bids = append(bidResponse.Bids, b)
+	}
+
+	if len(request.Cur) == 0 {
+		bidResponse.Currency = "USD"
+	}
+
+	return bidResponse, errs
+}
+
+func getBidFromResponse(bidResponse *resetDigitalBid) (*openrtb2.Bid, error) {
+
+	bid := &openrtb2.Bid{
+		ID:    bidResponse.BidID,
+		Price: bidResponse.CPM,
+		ImpID: bidResponse.ImpID,
+		CID:   bidResponse.CID,
+		CrID:  bidResponse.CrID,
+		AdM:   bidResponse.HTML,
+	}
+
+	w, err := strconv.ParseInt(bidResponse.W, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	bid.W = w
+
+	h, err := strconv.ParseInt(bidResponse.H, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	bid.H = h
+	return bid, nil
+}
+
+func getBidType(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
+	if imp.Banner != nil {
+		return openrtb_ext.BidTypeBanner, nil
+	} else if imp.Video != nil {
+		return openrtb_ext.BidTypeVideo, nil
+	} else if imp.Audio != nil {
+		return openrtb_ext.BidTypeAudio, nil
+	}
+
+	return "", fmt.Errorf("failed to find matching imp for bid %s", imp.ID)
+}
+
+func GetMediaTypeForImp(reqImps map[string]openrtb2.Imp, bidImpID string) (openrtb_ext.BidType, error) {
+	mediaType := openrtb_ext.BidTypeBanner
+
+	if reqImp, ok := reqImps[bidImpID]; ok {
+		if reqImp.Banner == nil && reqImp.Video != nil {
+			mediaType = openrtb_ext.BidTypeVideo
+		} else if reqImp.Audio != nil {
+			mediaType = openrtb_ext.BidTypeAudio
+		}
```

Comment:
I'm a bit confused on the precedence here. It looks like your order of precedence might be banner, video, audio but I'm not sure. I'm confused by the fact that this line doesn't check if banner is nil:
`} else if reqImp.Banner == nil && reqImp.Audio != nil {`

You could probably make this more straightforward by restructuring this function with early returns (untested):
```
if reqImp, ok := reqImps[bidImpID]; ok {
	if reqImp.Banner != nil {
		return openrtb_ext.BidTypeBanner, nil
	}
	if reqImp.Video != nil {
		return openrtb_ext.BidTypeVideo, nil
	}
	if reqImp.Audio != nil {
		return openrtb_ext.BidTypeAudio, nil
	}
}
return "", fmt.Errorf("unknown media type for bid imp ID %s", bidImpID)
```